internal/commands/user/list/command.go

```diff
@@ -13,8 +20,129 @@ var Command = cli.CommandDefinition{
 	Command:     &command{},
 }
 
-type command struct{}
+type command struct {
+	inputs      inputs
+	realmClient realm.Client
+	users       []realm.User
+}
+
+func (cmd *command) Flags(fs *pflag.FlagSet) {
+	cmd.inputs.Flags(fs)
+
+	fs.VarP(&cmd.inputs.UserState, flagState, flagStateShort, flagStateUsage)
+	fs.BoolVarP(&cmd.inputs.Pending, flagStatus, flagStatusShort, false, flagStatusUsage)
+	fs.StringSliceVarP(&cmd.inputs.ProviderTypes, flagProviderTypes, flagProviderTypesShort, []string{}, flagProviderTypesUsage)
+	fs.StringSliceVarP(&cmd.inputs.Users, flagUsers, flagUsersShort, []string{}, flagUsersUsage)
+}
+
+func (cmd *command) Inputs() cli.InputResolver {
+	return &cmd.inputs
+}
+
+func (cmd *command) Setup(profile *cli.Profile, ui terminal.UI) error {
+	cmd.realmClient = realm.NewAuthClient(profile.RealmBaseURL(), profile.Session())
+	return nil
+}
 
 func (cmd *command) Handler(profile *cli.Profile, ui terminal.UI) error {
+	app, appErr := cli.ResolveApp(ui, cmd.realmClient, cmd.inputs.Filter())
+	if appErr != nil {
+		return appErr
+	}
+	users, usersErr := cmd.realmClient.FindUsers(
+		app.GroupID,
+		app.ID,
+		realm.UserFilter{
+			State:     cmd.inputs.UserState,
+			Pending:   cmd.inputs.Pending,
+			Providers: cmd.inputs.ProviderTypes,
+			IDs:       cmd.inputs.Users,
+		},
+	)
+	if usersErr != nil {
+		return fmt.Errorf("failed to list users: %s", usersErr)
+	}
+
+	cmd.users = users
+
 	return nil
 }
+
+const (
+	headerID                     = "ID"
+	headerName                   = "Name"
+	headerEmail                  = "Email"
+	headerEnabled                = "Enabled"
+	headerType                   = "Type"
+	headerLastAuthenticationDate = "Last Authentication"
+)
+
+func (cmd *command) Feedback(profile *cli.Profile, ui terminal.UI) error {
+	if len(cmd.users) == 0 {
+		return ui.Print(terminal.NewTextLog("No available users to show"))
+	}
+	var usersByProvider = make(map[string][]realm.User, 0)
+	for _, user := range cmd.users {
+		for _, identity := range user.Identities {
+			usersByProvider[identity.ProviderType] = append(usersByProvider[identity.ProviderType], user)
+		}
+	}
+
+	var logs []terminal.Log
+	for provider, users := range usersByProvider {
+		sort.Slice(
+			users,
+			func(i, j int) bool {
```

Comment:
let's extract this `func` out to an unexported one in this file, something like `sortUsersByLastAuthentication` or something